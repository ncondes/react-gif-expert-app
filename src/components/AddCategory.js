import { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({ setCategories }) => {

  const [ inputValue, setInputValue ] = useState('');

  // handleInputChange
  const handleInputChange = e => setInputValue( e.target.value );
  

  // handleSubmit
  const handleSubmit = e => {
    e.preventDefault();

    if ( inputValue.trim().length > 2 ) {
      // para no tener que recibir la prop de category, podemos usar una funcion flecha para recibir el valor previo que se tenia de estado en categories.
      setCategories( prevCategories => [ inputValue, ...prevCategories ] );
      setInputValue('');
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <input
        type='text'
        value={ inputValue }
        onChange={ handleInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}

