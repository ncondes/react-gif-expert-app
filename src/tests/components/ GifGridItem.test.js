import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Pruebas en GifGridItem.js', () => {

  const title = 'title';
  const url = 'url'
  const wrapper = shallow( <GifGridItem 
    title={ title }
    url={ url }
  /> )

  test('Debe de mostrar el componente correctamente', () => {

    expect( wrapper ).toMatchSnapshot();

  });

  // Este text se realiza para saber si el parrafo en el componente tiene como contenido -> title
  test('Debe de tener un parrafo con el title', () => {
    
    const parrafo = wrapper.find('p');
    expect( parrafo.text().trim() ).toBe( title );

  });

  test('Debe de tener la imagen igual al url y alt de los props, lo mismo con el title', () => {
    
    const img = wrapper.find('img');
    // para saber el elemento se usa .html()
    // para saber las props del elemento se usa .props()
    // o si se desea conocer una de manera individual se usa .prop('property') => ejemplo: img.prop('src')
    // console.log( img.props().src );
    
    expect( img.prop( 'src' )).toBe( url );
    expect( img.prop( 'alt' )).toBe( title );

  });
  
  test('Deberia de tener la clase animate__fadeInUp', () => {
    
    const div = wrapper.find('div');

    // 2 formas de hacerlo:
    // expect( div.prop( 'className' ).includes( 'animate__fadeInUp' )).toBe( true );
    expect( div.hasClass( 'animate__fadeInUp' )).toEqual( true );

  });
  
  

})