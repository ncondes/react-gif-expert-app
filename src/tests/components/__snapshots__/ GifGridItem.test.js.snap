// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem.js Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="title"
    src="url"
  />
  <p>
    title
  </p>
</div>
`;
